### PR TITLE
chore: Add DocumentFragment to Container type

### PR DESCRIPTION
**Pull request by eps1lon**
## Summary

Accept `DocumentFragment` as a `container` in `createRoot` i.e. sync `Container` type with runtime `isValidContainer` implementation:
https://github.com/facebook/react/blob/7ec4c55971aad644616ca0b040f42410659fe802/packages/react-dom/src/client/ReactDOMRoot.js#L224-L231

Brings clarity about the valid container types for TypeScript typings. It's always a bit sketchy to refer to runtime behavior especially if flow types don't match the runtime.

## Test Plan

- [x] CI green.

### CODE REVIEW

File: packages/react-dom/src/client/ReactDOMComponent.js [hunk 1]
```diff
@@ -243,7 +243,7 @@ if (__DEV__) {
 }
 
 function getOwnerDocumentFromRootContainer(
-  rootContainerElement: Element | Document,
+  rootContainerElement: Element | Document | DocumentFragment,
```
Comment (eps1lon) [inline packages/react-dom/src/client/ReactDOMComponent.js:246]: Not sure whether this file intentionally inlines the container type or if we should replace this with 
```suggestion
  rootContainerElement: Container,
```
from `ReactDOMHostConfig`

**Comment by gaearon** [comment]
We support that?

**Comment by eps1lon** [comment]
> We support that?

At least you explicitly don't warn at at runtime: https://github.com/facebook/react/blob/7ec4c55971aad644616ca0b040f42410659fe802/packages/react-dom/src/client/ReactDOMRoot.js#L224-L231

**Comment by eps1lon** [comment]
@gaearon We frequently get requests to add DocumentFragment to the TypeScript types but it's not clear from the React side (no documentation, flow types don't match, runtime type checking does match) whether `DocumentFragment` is a valid container or not.

Would be nice to get some official confirmation that I can defer to.

**Comment by eps1lon** [comment]
Fixed by https://github.com/facebook/react/pull/24110

**Pull request closed without merging**